### PR TITLE
feat(material/core): use strong focus indicators in high contrast mode

### PR DESCRIPTION
**Pull request by crisbeto**
Reuses the strong focus indicators styling for high-contrast mode users.

### CODE REVIEW

File: src/material-experimental/mdc-list/list.scss [hunk 1]
```diff
@@ -58,9 +58,17 @@
 // The MDC-based list items already use the `::before` pseudo element for the standard
 // focus/selected/hover state. Hence, we need to have a separate list-item spanning
 // element that can be used for strong focus indicators.
-.mat-mdc-list-item > .mat-mdc-focus-indicator {
-  @include mat.private-fill();
-  pointer-events: none;
+.mat-mdc-list-item {
+  & > .mat-mdc-focus-indicator {
```
Comment (devversion) [inline src/material-experimental/mdc-list/list.scss:62]:
```suggestion
  > .mat-mdc-focus-indicator {
```

File: src/material-experimental/mdc-radio/radio.scss [hunk 2]
```diff
@@ -70,6 +69,12 @@
     .mat-ripple-element {
       opacity: mdc-radio-theme.$ripple-opacity;
     }
+
+    // Radio components have to set `border-radius: 50%` in order to support density scaling
+    // which will clip a square focus indicator so we have to turn it into a circle.
+    &::before {
```
Comment (devversion) [inline src/material-experimental/mdc-radio/radio.scss:75]: I saw this a couple of times now. OOC, is this strictly related to the focus indicators? You mention density scaling, that's why I ask.

**Review by zarend — APPROVED** [review]

**Review by devversion — APPROVED** [review]
LGTM, pretty nice!

File: src/material-experimental/mdc-radio/radio.scss (hunk 2 above)
Comment (crisbeto) [reply src/material-experimental/mdc-radio/radio.scss:75]: I didn't write it, but I think that it's referring to the fact that it uses percentages instead of pixels for the radius.
